src/middlewares/auth.js

```diff
@@ -1,59 +1,61 @@
-/* eslint-disable prefer-destructuring */
+import { config } from 'dotenv';
 import jwt from 'jsonwebtoken';
 
-const auth = {
-  generateToken(user) {
+config();
+
+export default {
+
+  generateToken(newUser) {
+    const {
+      id, first_name, last_name, email, role
+    } = newUser;
     return jwt.sign({
-      id: user.id,
-      role: user.role,
-    }, process.env.SECRET, { expiresIn: '7d' });
+      id,
+      first_name,
+      last_name,
+      email,
+      role
+    }, process.env.JWT_SECRET, { expiresIn: '7d' });
   },
 
-  verifyToken(token) {
-    let decoded = {};
-    try {
-      decoded.payload = jwt.verify(token, process.env.SECRET);
-    } catch (error) {
-      decoded = { error: error.message };
+  checkToken: (req, res, next) => {
+    const { authorization } = req.headers;
+    let token;
+
+
+    if (typeof authorization !== 'undefined' && authorization.includes('Bearer')) {
```

Comment:
Great job here. I think for sake of clarity and consistency, we should just stick with `req.headers.authorization` with `Bearer` prepended.

Reply:
Well; you never can be too sure regarding `the mexicans` lol.
Always try to cover the `edge cases`.

Reply:
That's why we have a documentation. They should read it.

Reply:
`The mexicans` are not the engineers or anybody on the team; or anybody who cared to read the documentation. They are people outside, who would want to abuse your code or `test your thinking faculty`, LOL.
Don't give them that chance !!!

Reply:
That's why we why we have validations. 
If they don't send the token via the `req.headers.authorization` it should return a 403 status. So I don't see how that's a problem.

Reply:
Let me put it this way. If you want to login to google email account, you could do: sampleemail@gmail.com , you can do sampleemail@googlemail.com or you can just type samplemail, they ALL work. what you're asking is like restricting users to just typing `samplemail@gmail.com` and rejecting any other thing telling them to go read the documentation. You code has to be forebearing whenever it can allow it without breaking the product. You dig?